### PR TITLE
Refactor: test CollectionUtils -> general collectionUtils. 

### PR DESCRIPTION
**Pull request by Hweinstock**
## Problem
The difference between the collection utils and the general collection utils is blurry and not well defined. While some utility functions are only used in testing, this is not obvious. Follow up from https://github.com/aws/aws-toolkit-vscode/pull/3534#discussion_r1222241097. 
## Solution
Move all of the utility functions regarding collections into one place, so that they don't get lost. 

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/shared/utilities/collectionUtils.ts [hunk 1]
```diff
@@ -488,3 +488,25 @@ export async function* joinAll<T>(iterable: AsyncIterable<AsyncIterable<T>>): As
         }
     } while (!iterables.completed)
 }
+
+// TODO: what is the point of this? should it live in src/shared/utilities/collectionUtils.ts ?
```
Comment (JadenSimon) [inline src/shared/utilities/collectionUtils.ts:492]:
```suggestion

```

**Review by JadenSimon — APPROVED** [review]

**Comment by JadenSimon** [comment]
Target `master`?

**Comment by Hweinstock** [comment]
This branch also includes the changes for the EC2 prompter. I'll remove the changes from here regarding the EC2 prompter, then can target master with this PR.

**Comment by Hweinstock** [comment]
Think I removed any remnants from ec2. However, first time using `rebase` so lmk if I messed something up.

File: src/shared/utilities/collectionUtils.ts [hunk 2]
```diff
@@ -488,3 +488,24 @@ export async function* joinAll<T>(iterable: AsyncIterable<AsyncIterable<T>>): As
         }
     } while (!iterables.completed)
 }
+
+export async function* asyncGenerator<T>(items: T[]): AsyncIterableIterator<T> {
```
Comment (justinmk3) [inline src/shared/utilities/collectionUtils.ts:492]: looks like the `test/utilities/collectionUtils.ts` file still exists?

**Pull request merged**